lib/shopify_app/managers/webhooks_manager.rb

```diff
@@ -38,18 +40,37 @@ def add_registrations
         return unless ShopifyApp.configuration.has_webhooks?
 
         ShopifyApp.configuration.webhooks.each do |attributes|
+          webhook_path = path(attributes)
+
           ShopifyAPI::Webhooks::Registry.add_registration(
             topic: attributes[:topic],
             delivery_method: attributes[:delivery_method] || :http,
-            path: attributes[:path] || attributes[:address],
-            handler: webhook_job_klass(attributes[:path]),
+            path: webhook_path,
+            handler: webhook_job_klass(webhook_path),
             fields: attributes[:fields]
           )
         end
       end
 
       private
 
+      def path(webhook_attributes)
+        path = webhook_attributes.dig(:path)
+        address = webhook_attributes.dig(:address)
+        uri = URI(address) if address
+
+        # Fail unless path or address are present
+        if path.nil? && address.nil?
+          raise ShopifyApp::MissingWebhookJobError("The :path attribute is required for webhook registration.")
```

Comment:
```suggestion
          raise ShopifyApp::MissingWebhookJobError, "The :path attribute is required for webhook registration.")
```
I'm not sure if they are exactly equivalent but I think this is the usual form?